presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -322,35 +312,31 @@ private static Time toSqlTime(LocalTime localTime)
         return new Time(Time.valueOf(localTime).getTime() + NANOSECONDS.toMillis(localTime.getNano()));
     }
 
-    public static ColumnMapping timeColumnMapping(ConnectorSession session)
+    public static ColumnMapping timeColumnMapping()
     {
-        if (session.isLegacyTimestamp()) {
-            ZoneId sessionZone = ZoneId.of(session.getTimeZoneKey().getId());
-            return ColumnMapping.longMapping(
-                    TIME,
-                    (resultSet, columnIndex) -> {
-                        LocalTime time = resultSet.getObject(columnIndex, LocalTime.class);
-                        return toPrestoLegacyTimestamp(time.atDate(LocalDate.ofEpochDay(0)), sessionZone);
-                    },
-                    timeWriteFunction(session));
-        }
+        return ColumnMapping.longMapping(
+                TIME,
+                (resultSet, columnIndex) -> {
+                    LocalTime time = resultSet.getObject(columnIndex, LocalTime.class);
+                    return (roundDiv(time.toNanoOfDay(), NANOSECONDS_PER_MILLISECOND) * PICOSECONDS_PER_MILLISECOND) % PICOSECONDS_PER_DAY;
+                },
```

Comment:
This expression is only used in one place. It also has very specific semantics that I don't think apply in general (they are an artifact of JDBC connectors only supporting `time(3)`). Essentially it converts nanoseconds to milliseconds with rounding and then scales to picoseconds to match the internal representation expected by the engine.

Reply:
That's fair. I do still think this should be exposed as `toPrestoTime` (possibly along with `truncatingToPrestoTime`); those are both useful for implementing new JDBC connectors (and it would make #4558 completely unnecessary).

(There were two uses of this originally; the other is the one that used to go through `toLocalTime`, and I misread the truncating version as a third, oops.)